src/main/java/tutorlink/command/ListComponentCommand.java

```diff
@@ -0,0 +1,27 @@
+package tutorlink.command;
+
+import java.util.HashMap;
+
+import tutorlink.appstate.AppState;
+import tutorlink.exceptions.TutorLinkException;
+import tutorlink.result.CommandResult;
+
+public class ListComponentCommand extends Command {
+
+    public static final String COMMAND_WORD = "list_component";
+    private static final String SUCCESS_MESSAGE = "Here are your components:";
+    private static final String EMPTY_MESSAGE = "No components found.";
+
+    @Override
+    public CommandResult execute(AppState appState, HashMap<String, String> parameters) throws TutorLinkException {
```

Comment:
Is there a reason why CommandResult does not throw TutorLinkException despite its function definition stating otherwise?